Src/AutoFixture/Kernel/ArrayRelay.cs

```diff
@@ -48,17 +49,21 @@ public object Create(object request, ISpecimenContext context)
             var elements = specimen as IEnumerable;
             if (elements == null)
                 return new NoSpecimen();
-            return ArrayRelay.ToArray(elements, elementType);
+            return ToArray(elements, elementType);
         }
 
-        private static object ToArray(IEnumerable e, Type elementType)
+        private static object ToArray(IEnumerable elements, Type elementType)
         {
-            var al = new ArrayList();
-            foreach (var element in e)
+            var collection = elements as ICollection;

```

Comment:
`Enumerable.Count` [already does that.](http://referencesource.microsoft.com/#System.Core/System/Linq/Enumerable.cs,1191)